### PR TITLE
feat(TCCUI65/calendar): show 6 weeks

### PR DESCRIPTION
**Pull request by jillyan**
**What is the problem this PR is trying to solve?**
TCCUI65: The current date calendar shows different number of weeks depending on days of month.
**What is the chosen solution to this problem?**
Keep 6 weeks in date calendar, full fill the calendar with date in previous month and later month.
http://guidelines.talend.com/document/92132#/ui-controls/date-time-picker-1561102408
**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/__mocks__/day-calendar.js [hunk 1]
```diff
@@ -24,7 +28,7 @@ class DayCalendar extends React.Component {
 								const day = getDate(date);
 								return (
 									<td key={dayIndex}>
-										<button onKeyDown={event => onKeyDown(event, this.calendarRef, day - 1)}>
+										<button id={this.isCurrentMonth(date) && day} onKeyDown={event => onKeyDown(event, this.calendarRef, day - 1)}>
```
Comment (jmfrancois) [inline packages/components/__mocks__/day-calendar.js]: this id look very wick, and even if we add prefix to it like `week-1` It means you can t have multiple calendar on the same page. Don t know if this acceptable.

Comment (jillyan) [reply packages/components/__mocks__/day-calendar.js]: make sense, will change it.

Comment (jillyan) [reply packages/components/__mocks__/day-calendar.js]: use name for now

File: packages/components/src/DateTimePickers/pickers/MonthPicker/MonthPicker.component.js [hunk 2]
```diff
@@ -54,6 +54,7 @@ class MonthPicker extends React.PureComponent {
 								return (
 									<td key={index} {...tdProps}>
 										<button
+											id={index}
```
Comment (jmfrancois) [inline packages/components/src/DateTimePickers/pickers/MonthPicker/MonthPicker.component.js]: same as previous

Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/MonthPicker/MonthPicker.component.js]: updated

File: output/components.eslint.txt [hunk 3]
```diff
@@ -19,6 +19,9 @@ The react/require-extension rule is deprecated. Please use the import/extensions
   33:11  error  Caution: `i18next` also has a named export `t`. Check if you meant to write `import {t} from 'i18next'` instead  import/no-named-as-default-member
   35:11  error  Caution: `i18next` also has a named export `t`. Check if you meant to write `import {t} from 'i18next'` instead  import/no-named-as-default-member
 
+/home/travis/build/Talend/ui/packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.component.js
+  13:8  error  'isWithinRange' is defined but never used  no-unused-vars
```
Comment (jsomsanith-tlnd) [inline output/components.eslint.txt]: Fix this error please

File: packages/components/__mocks__/month-calendar.js [hunk 4]
```diff
@@ -13,46 +13,46 @@ class MonthCalendar extends React.Component {
 				<tbody>
 					<tr>
 						<td>
-							<button onKeyDown={event => onKeyDown(event, this.calendarRef, 0)}>JAN</button>
+							<button name='0' onKeyDown={event => onKeyDown(event, this.calendarRef, 0)}>JAN</button>
```
Comment (jsomsanith-tlnd) [inline packages/components/__mocks__/month-calendar.js]: This is not valid jsx, please use double quotes

File: packages/components/src/DateTimePickers/pickers/MonthPicker/MonthPicker.component.js [hunk 5]
```diff
@@ -54,6 +54,7 @@ class MonthPicker extends React.PureComponent {
 								return (
 									<td key={index} {...tdProps}>
 										<button
+											name={index}
```
Comment (jsomsanith-tlnd) [inline packages/components/src/DateTimePickers/pickers/MonthPicker/MonthPicker.component.js]: Why did you added this name attribute in month picker ?

File: packages/components/src/DateTimePickers/generator.js [hunk 6]
```diff
@@ -35,17 +35,20 @@ export function buildDayNames(firstDayOfweek = 1, t) {
 /**
  * Generate the set of weeks for a specific month
  */
-export function buildWeeks(year, monthIndex, firstDayOfWeek = 1) {
+export function buildWeeks(year, monthIndex, firstDayOfWeek = 1, show6weeks = false) {
```
Comment (jsomsanith-tlnd) [inline packages/components/src/DateTimePickers/generator.js]: I don't think this is useful : `show6weeks`
The only UI is with 6 weeks

File: packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.component.js [hunk 7]
```diff
@@ -92,63 +105,61 @@ class DatePicker extends React.PureComponent {
...
+									});
+								}
+								if (today) {
+									ariaLabel = t('DATEPICKER_DAY_TODAY', {
+										defaultValue: 'Today, {{date}}',
+										date: ariaLabel,
+									});
 								}
-								return <td key={j} />;
+								const buttonProps = this.isCurrentMonth(date) ? { name: day } : undefined;
```
Comment (jsomsanith-tlnd) [inline packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.component.js]: Can you change the `name attribute` to a data-attribute which is more standard to add custom info.
For example `data-day`

File: packages/components/src/DateTimePickers/generator.js (hunk 6 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/generator.js]: but with the parameter, user could show different view if he like does not like to show 6 weeks.

File: packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.component.js (hunk 7 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.component.js]: name is used for the key action, in code 
withCalendarGesture.js
function getAllItems(calendarRef) {
	return calendarRef.querySelectorAll('td > button[name]');
}
and this is used for date picker and month picker?
How about 'data-valid'?

File: packages/components/src/DateTimePickers/pickers/MonthPicker/MonthPicker.component.js (hunk 5 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/MonthPicker/MonthPicker.component.js]: for the navigation by keyboard.
function getAllItems(calendarRef) {
	return calendarRef.querySelectorAll('td > button[name]');
}
and as your previous comment, I think about use 'data-valid' to replace 'name'

File: packages/components/src/DateTimePickers/generator.js (hunk 6 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/generator.js]: yes, you are right. will update it.

File: packages/components/__mocks__/month-calendar.js (hunk 4 above)
Comment (jillyan) [reply packages/components/__mocks__/month-calendar.js]: fixed

File: output/components.eslint.txt (hunk 3 above)
Comment (jillyan) [reply output/components.eslint.txt]: fixed

File: packages/components/src/DateTimePickers/pickers/MonthPicker/MonthPicker.component.js (hunk 5 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/MonthPicker/MonthPicker.component.js]: changed to be 'data-value'

File: packages/components/src/DateTimePickers/generator.js (hunk 6 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/generator.js]: show6weeks is removed

File: packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.component.js (hunk 7 above)
Comment (jillyan) [reply packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.component.js]: changed to be 'data-value'

**Pull request merged**